### ISSUE TITLE
Migrate "Channel.Save" to Sync by default

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#10769]
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15354)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `Save` in the `Channel` store.

The expected way to implement it is, go to the `Channel` store implementation in the `store/sqlstore/` directory, modify the method `Save` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

**Comment by gupsho** [comment]
I can take this.

**Comment by andresoro** [comment]
I have some free time on my hands so I can begin working on this.

**Comment by hanzei** [comment]
Thanks @andresoro!

**Comment by andresoro** [comment]
@jespino How should I handle the changes to the the Channel().Save() method with store.Must()? My initial idea is to do something like: 
```
ch, err := ss.Channel().Save(&channel, -1)
storeCh := make(store.StoreChannel, 1)
storeCh <- store.StoreResult{
	Data: ch,
	Err:  err,
}
channel = *store.Must(storeCh).(*model.Channel)

```

**Comment by jespino** [comment]
The store.Must is a solution to deal with the go channel and the result struct. But without the channel and the struct you just simply have to check if there is an error. The solution would be:

```go
channel, err := ss.Channel().Save(&channel, -1)
require.Nil(t, err)
```

**Comment by andresoro** [comment]
Going to reopen when I get all test cases passing.

**Comment by jespino** [comment]
Fixed by #10871
